src/mdframe/reader.py

```diff
@@ -39,38 +42,50 @@ def data_dir_to_dataframes(data_dir_path: Path,
                            schema: Dict) -> List[pd.DataFrame]:
     metadata_file_paths = data_dir_path.glob(f"*.{metadata_file_extension}")    
     metadata_file_contents = [load_metadata_file(path, schema) for path in metadata_file_paths]
-    metadata_records = map(metadata_file_to_df, metadata_file_contents)
-    return list(metadata_records)
 
+    # remove all None(s) representing empty reads
+    metadata_file_contents = [text for text in metadata_file_contents if text != None]
+    
+    metadata_records = []
+    for text in metadata_file_contents:
+        metadata_records.append(pd.Series(text))
+
+    return metadata_records
+    
 @dataclass
 class Config:
     data_dir_path: Path
     metadata_file_extension: Optional[MetadataFileExtension]
     # data_file_extensions: List[DataFileExtension]
-    schema_url: Url | Path
+    schema_loc: Path | str
     __schema: Dict | None = None
 
     @property
     def schema(self):
         if self.__schema is None:
-            if isinstance(self.schema_url, Path):
-                if not self.schema_url.exists():
-                    raise ValueError(f"Path {self.schema_url} does not seem to point to a valid JSON schema file")
+            if isinstance(self.schema_loc, Path):
+                if not self.schema_loc.exists():
+                    raise ValueError(f"Path {self.schema_loc} does not seem to point to a valid JSON schema file")
 
-                with open(self.schema_url, "r") as f:
+                with open(self.schema_loc, "r") as f:
                     self.__schema = json.loads(f.read())
-            else: # schema_url is of type Url
+            else: # schema_url is a string (representing a URL)
                 http = urllib3.PoolManager()
-                response = http.request('GET', self.schema_url)
+                response = http.request('GET', self.schema_loc)
 
                 if response.status == 200:
                    self.__schema = json.loads(response.data.decode('utf-8'))
                 else:
-                    raise HTTPError(f"Failed to fetch schema from URL {self.schema_url}")
+                    raise HTTPError(f"Failed to fetch schema from URL {self.schema_loc}")
         return self.__schema
 
+
 def run(config: Config):
-    return data_dir_to_dataframes(**config.__dict__)
+    return data_dir_to_dataframes(
```

Comment:
Not sure why you changed the config dict unpack to explicit kwargs? If there are kwargs unused by the `data_dir_to_dataframes` function, you could add `**kwargs` to the signature of that function, then kwargs unpacking would simply pass on the rest of the params as `kwargs` and they would simply remain unused. I personally prefer simpler function signatures.

Reply:
I think I did is because of the way `_schema` is defined for the Config class. If I just pass in the dictionary as is, `_schema` is still equal to None because at the time of its evaluation, the `schema` method hasn't been called a single time.

So to counteract this, I decided to unpack the dictionary and explicitly make a call to `config.schema` and pass this in.